medcat/compare_models/model_comparison.ipynb

```diff
@@ -0,0 +1,1504 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# This notebook will attempt to show how to compare two models\n",
+    "\n",
+    "We often arrive in a situation where we've got multiple different models. Yet we're note sure which one we should focus on or start from for a particular task.\n",
+    "This notebook aims to introduce some tools that (hopefully) help us do that."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Initial input - models and data\n",
+    "\n",
+    "There are two different workflows this notebook can handle:\n",
+    "1. Compare two different model packs\n",
+    "  - Provide 2 model pack paths\n",
+    "  - Provide a documents file\n",
+    "2. Compare model pack with and without supervised training\n",
+    "  - Provide 1 model pack path\n",
+    "  - Provide a file path to a MedCATtrainer (MCT) export\n",
+    "  - Provide a document file\n",
+    "\n",
+    "The model packs can be either the `.zip` file (which will be automatically unzipped) or the folder.\n",
+    "\n",
+    "The documents file is expected in a `.csv` format with two columns (`id`, and `text`).\n",
+    "\n",
+    "The MCT export is expected in the format given by MedCATtrainer.\n",
+    "\n",
+    "For the two approaches, there is a slightly different internal workflow.\n",
+    "But other than ticking the checkbox, the process should be identical to the user."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "af259b92622d4fd2b04ab64549e17715",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "FileChooser(path='/Users/martratas/Documents/CogStack/.MedCAT.nosync/working_with_cogstack/models/modelpack', …"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "c9edb3221d464cb7a2968d42595188fa",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "FileChooser(path='/Users/martratas/Documents/CogStack/.MedCAT.nosync/working_with_cogstack/models/modelpack', …"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "75fd5d8c615149ff8bc94f0fdacde303",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "FileChooser(path='/Users/martratas/Documents/CogStack/.MedCAT.nosync/working_with_cogstack/medcat/compare_mode…"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "from ipyfilechooser import FileChooser\n",
```

Comment:
I tried that earlier on the MedCAT tutorials and saw it was already included. So kind of assumed it would ship with the rest of the Jupyter stuff because I know it silently installs a bunch of stuff already.

But you're right, even _if_ it _sometimes_ ships with Jupyter Hub, it's best to explicitly declare it.